core/src/main/java/io/questdb/griffin/NetworkSqlExecutionCircuitBreaker.java

```diff
@@ -28,42 +28,66 @@
 import io.questdb.network.NetworkFacade;
 import io.questdb.std.MemoryTag;
 import io.questdb.std.Unsafe;
+import io.questdb.std.datetime.microtime.MicrosecondClock;
 
 import java.io.Closeable;
 
-public class HttpSqlExecutionInterruptor implements SqlExecutionInterruptor, Closeable {
+public class NetworkSqlExecutionCircuitBreaker implements SqlExecutionCircuitBreaker, Closeable {
```

Comment:
nit: the class name is a bit misleading. The [circuit breaker](https://martinfowler.com/bliki/CircuitBreaker.html) assumes that the newly received requests (queries in our case) get rejected immediately for a certain period of time after it enters the open state. See the FSM illustration here: https://martinfowler.com/bliki/images/circuitBreaker/state.png

This class doesn't look similar.

Reply:
it used to be called "interruptor" - i'm happy to accept name suggestions

Reply:
Interruptor sounds better IMO.

Reply:
Interruptor is not a word :)